packages/firestore/src/local/local_store.ts

```diff
@@ -899,8 +909,16 @@ class LocalStoreImpl implements LocalStore {
           .next((cached: TargetData | null) => {
             if (cached) {
               // This target has been listened to previously, so reuse the
-              // previous targetID.
+              // previous targetID if no `readFrom` is provided.
+              // If `readFrom` is provided, the cached target data needs to be
+              // compared and updated if `readFrom` is newer.
               // TODO(mcg): freshen last accessed date?
+              if (readFrom && cached.snapshotVersion.compareTo(readFrom) < 0) {
```

Comment:
On second thought, it seems unlikely that a bundle has a newer readTime than the target we have in cache. Maybe we can just go back to the old code.